shell/browser/electron_browser_handler_impl.cc

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2019 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/electron_browser_handler_impl.h"
+
+#include <utility>
+
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/render_process_host.h"
+#include "mojo/public/cpp/bindings/self_owned_receiver.h"
+
+namespace electron {
+ElectronBrowserHandlerImpl::ElectronBrowserHandlerImpl(
+    content::RenderFrameHost* frame_host)
+    : render_process_id_(frame_host->GetProcess()->GetID()),
+      render_frame_id_(frame_host->GetRoutingID()),
+      weak_factory_(this) {
+  content::WebContents* web_contents =
+      content::WebContents::FromRenderFrameHost(frame_host);
+  content::WebContentsObserver::Observe(web_contents);
+}
+
+ElectronBrowserHandlerImpl::~ElectronBrowserHandlerImpl() = default;
+
+void ElectronBrowserHandlerImpl::BindTo(
+    mojo::PendingReceiver<mojom::ElectronBrowser> receiver) {
+  // Note: BindTo might be called for multiple times.
```

Comment:
I'm confused what would result in this method being called multiple times. I can only see a single reference to it, from `ElectronBrowserHandlerImpl::Create`, which would mean it's called exactly once per instance.

And in which case, we should pass this to the constructor.

Reply:
ooops... my bad... I copied this from https://github.com/electron/electron/blob/master/shell/renderer/electron_api_service_impl.cc -- I'll remove it